### PR TITLE
Mbio eda feature site

### PR DESCRIPTION
**Pull request by asizemore**

### CODE REVIEW

**Comment by asizemore** [comment]
Hey @dmfalke and @jtlong3rd ! Making this PR in hopes of figuring out what needs to be done in order to get this branch finally merged. Importantly, I learned that microbiome is _not_ freezing the live site for at least one more build cycle. Is it possible to merge this branch and still have qa and live use the legacy code and the feature site use the eda? My biggest concern is in setting the configuration vars in `default.yml`. Though I suppose all the wrapping we do in `main.js` could cause some issues, too...

Anyways, if there's no reasonable way to get this branch merged while we still have development happening on the wdk site, i'm not opposed to maintaining the branch a little longer. Curious what you both think!

**Comment by dmfalke** [comment]
As it appears that this branch has the latest from `master`, you should only need to set this value to `false`: https://github.com/VEuPathDB/MicrobiomeWebsite/blob/9f7b6341296180a1fe6808772f7f2777b55ba54a/Model/lib/conifer/roles/conifer/vars/Microbiome/default.yml#L13-L14.

We will then need to override this value for the feature site, which I believe can be done in the jenkins configuration.

It would be a good idea to first test this on a development site to make sure things work as expected. I'm happy to help with this effort.

**Review by jtlong3rd — COMMENTED** [review]
One potential "gotcha" with merging: the latest version of `@veupathdb/web-eda` has a `peerDependency` of `@veupathdb/wdk-client: ^0.8.1`, so just to be safe, we'd want to try upgrading `@veupathdb/wdk-client` too.

(The breaking change in `@veupathdb/wdk-client: ^0.8.x` related to the new snackbar notifications, which appear in one spot on legacy microbiome: the strategy workspace. To fully accommodate that breaking change, you'll need to wire up the notifications by wrapping the `Page` in [these two components, similarly to what I did for genomics](https://github.com/VEuPathDB/ApiCommonWebsite/pull/61/files#diff-79cd0ec456b72dbf4200fc29ab937adb96d15d20a084044a29a7a1c802f6ca4dR190-R191). I'm happy to do this on your behalf, or let you take a stab at it. Your choice!)

File: Site/webapp/wdkCustomization/js/client/main.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
 import wrapWdkService from './wrapWdkService';
 import { wrapRoutes } from './routes';
-
+import { wrapWdkDependencies } from '@veupathdb/study-data-access/lib/shared/wrapWdkDependencies';
 import 'site/css/MicrobiomeSite.css';
 
 // Initialize the application.
 initialize({
   componentWrappers,
+  wrapWdkDependencies: partial(wrapWdkDependencies, edaServiceUrl),
```
Comment (jtlong3rd) [inline Site/webapp/wdkCustomization/js/client/main.js]: One defensive suggestion for accommodating legacy mbio and eda mbio:

```suggestion
  wrapWdkDependencies: partial(
    wrapWdkDependencies,
    useEda
      ? edaServiceUrl 
      : undefined
  ),
```

File: Site/webapp/wdkCustomization/js/client/component-wrappers/Page.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    return (
+           <UIThemeProvider
+              theme={{
+                palette: {
+                  primary: { hue: colors.mutedBlue, level: 500 },
+                  secondary: { hue: colors.mutedRed, level: 500 },
+                },
+              }}
+            >
+              <DefaultComponent {...props} />
```
Comment (jtlong3rd) [inline Site/webapp/wdkCustomization/js/client/component-wrappers/Page.tsx]: You could put the snackbar-related components here, in between the `UIThemeProvider` and the `DefaultComponent`.

**Comment by asizemore** [comment]
Thank you both so much! I'll give it a try and keep you posted!

**Comment by asizemore** [comment]
Update: all the changes you both suggested (thank you!) work for the eda. The only one you don't see pushed is changing the eda enabled var to 'false'. I've changed this locally and confirmed the override from `conifer_site_vars.yml` works, but am waiting to push this last change until I'm sure the feature site has the configured var. Otherwise when the feature site builds tonight it'll break!

When i've tested all changes with my local dev site, the site builds and loads but with no data because of an error about missing the `isPublic` attribute. I sent a slack [message](https://epvb.slack.com/archives/CBLPK9ZD3/p1657823349213259) about it and hopefully I can just fix it in the presenters like we did last week.

File: Site/webapp/wdkCustomization/js/client/component-wrappers/Page.tsx [hunk 3]
```diff
@@ -1,9 +1,12 @@
-import React from 'react';
+import React, { useMemo } from 'react';
```
Comment (jtlong3rd) [inline Site/webapp/wdkCustomization/js/client/component-wrappers/Page.tsx]:
```suggestion
import React from 'react';
```

File: Site/webapp/wdkCustomization/js/client/component-wrappers/Page.tsx [hunk 4]
```diff
@@ -22,8 +27,31 @@ export function Page(DefaultComponent: React.ComponentType<Props>) {
...
 }
+
+
+function translateNotificationsOnTop() {
+  return {
+    transform: 'translateY(84px)'
+  };
+}
+
+const VEuPathDBSnackbarProvider = makeSnackbarProvider(
```
Comment (jtlong3rd) [inline Site/webapp/wdkCustomization/js/client/component-wrappers/Page.tsx]:
```suggestion
const MicrobiomeSnackbarProvider = makeSnackbarProvider(
```

File: Site/webapp/wdkCustomization/js/client/component-wrappers/Page.tsx [hunk 5]
```diff
@@ -22,8 +27,31 @@ export function Page(DefaultComponent: React.ComponentType<Props>) {
                 },
               }}
             >
-              <DefaultComponent {...props} />
+              <VEuPathDBSnackbarProvider styleProps={snackbarStyleProps}>
```
Comment (jtlong3rd) [inline Site/webapp/wdkCustomization/js/client/component-wrappers/Page.tsx]:
```suggestion
              <MicrobiomeSnackbarProvider styleProps={snackbarStyleProps}>
```

**Review by jtlong3rd — APPROVED** [review]

**Comment by asizemore** [comment]
Update: the isPublic fix worked so the only outstanding question is where/when to set the eda enabled flag. Because we're pretty sure beta is prod (feature may be prod or dev not sure), we'll have to ask Bob B to set a override somewhere other than one of the two `default.yml` files. He's on vacation until July 25 so I will ping him when he returns. 

Otherwise running the wdk or eda sites off of this branch looks good!

**Comment by dmfalke** [comment]
Great! I would go ahead and merge this soon, with the default to have eda disabled.

**Pull request merged**